src/browser/services/CoreBrowserService.ts

```diff
@@ -14,6 +14,6 @@ export class CoreBrowserService implements ICoreBrowserService {
   }
 
   public get isFocused(): boolean {
-    return document.activeElement === this._textarea && document.hasFocus();
+    return (document.activeElement===this._textarea || (document.activeElement.shadowRoot && document.activeElement.shadowRoot.activeElement===this._textarea)) && document.hasFocus();
```

Comment:
Seems reasonable. Here's an improvement which should be functionally equivalent:

```suggestion
    return (document.activeElement===this._textarea || document.activeElement.shadowRoot?.activeElement===this._textarea) && document.hasFocus();
```